File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> scoville, int K) {
    priority_queue<int, vector<int>, greater<>> pq;
    for(auto s: scoville) pq.push(s);
    int answer = 0;
    while(!pq.empty()) {
        int first = pq.top(); pq.pop();
        if(first >= K) return answer;
        if(!pq.empty()){
            int second = pq.top(); pq.pop();
            answer++;
            pq.push(first + second * 2);
        }
    }
    return -1;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville)
    result = 0
    
    while scoville:
        first = heapq.heappop(scoville)
        
        if first >= K:
            return result
        
        if scoville:
            heapq.heappush(scoville, first + heapq.heappop(scoville) * 2)
            result += 1
            
    return -1
